tests/common/helpers/platform_api/fan.py

```diff
@@ -0,0 +1,66 @@
+""" This module provides interface to interact with the fan of the DUT
+    via platform API remotely """
+
+import json
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def fan_api(conn, index, name, args=None):
+    if args is None:
+        args = []
+    conn.request('POST', '/platform/chassis/fan/{}/{}'.format(index, name), json.dumps({'args': args}))
+    resp = conn.getresponse()
+    res = json.loads(resp.read())['res']
+    logger.info('Executing fan API: "{}", index: {}, arguments: "{}", result: "{}"'.format(name, index, args, res))
+    return res
+#
+# Methods inherited from DeviceBase class
+#
+
+def get_name(conn, index):
+    return fan_api(conn, index,'get_name')
+
+
+def get_presence(conn, index):
+    return fan_api(conn, index,'get_presence')
+
+
+def get_model(conn, index):
+    return fan_api(conn, index,'get_model')
+
+
+def get_serial(conn, index):
+    return fan_api(conn, index,'get_serial')
+
+
+def get_status(conn, index):
+    return fan_api(conn, index,'get_status')
+#
+# Methods defined in fanBase class
+#
+
+def get_name(conn, index):
+    return fan_api(conn, index, 'get_name')
```

Comment:
This is defined in the DeviceBase class, not in the Fan class. It is a redefinition, since you already defined it on lines 22-23. Please remove.

Reply:
fixed